spec/taskBehaviors.js

```diff
@@ -0,0 +1,267 @@
+describe('Tasks', function() {
+    beforeEach(function() {
+        jasmine.Clock.useMockForTasks();
+    });
+
+    afterEach(function() {
+        // Check that task schedule is clear after each test
+        expect(ko.tasks.length()).toEqual(0);
+    });
+
+    it('Should run in next execution cycle', function() {
+        var runCount = 0;
+        ko.tasks.schedule(function() {
+            runCount++;
+        });
+        expect(runCount).toEqual(0);
+
+        jasmine.Clock.tick(1);
+        expect(runCount).toEqual(1);
+    });
+
+    it('Should run multiple times if added more than once', function() {
+        var runCount = 0;
+        var func = function() {
+            runCount++;
+        };
+        ko.tasks.schedule(func);
+        ko.tasks.schedule(func);
+        expect(runCount).toEqual(0);
+
+        jasmine.Clock.tick(1);
+        expect(runCount).toEqual(2);
+    });
+
+    it('Should run scheduled tasks in the order they were scheduled', function() {
+        var runValues = [];
+        var func = function(value) {
+            runValues.push(value);
+        };
+
+        ko.tasks.schedule(func.bind(null, 1));
+        ko.tasks.schedule(func.bind(null, 2));
+
+        jasmine.Clock.tick(1);
+        expect(runValues).toEqual([1,2]);
+    });
+
+    it('Should run tasks again if scheduled after a previous run', function() {
```

Comment:
I'm not sure this spec covers what its name seems to suggest, unless I'm misinterpreting.

The name suggests (to me) that it covers what happens when you schedule the same function instance more than once, but given that it uses fn.bind to construct the param for ko.tasks.schedule, it's a different function instance getting scheduled each time.

Should this be changed to pass the same function instance each time?